### PR TITLE
Feat: add nearby search

### LINKED ISSUE #118: Add nearby search

### PR DESCRIPTION
**Pull request by g0g05arui**
## Summary

In this PR I address the issue #118.
If the user has granted the necessary permissions, we will only search / filter the places in a small radius centered around the user location.
If the user has not granted location permissions, the default location will be used / no boundaries applied => it will show places all around the globe.
If the user has granted permissions but has location disabled, a dialog will pop up to request the user to enable the location in the settings

## Changes

- **Repositories:** Updated the repository for places ( and googlesearch repo) to allow an optional location parameter
- **Bug Fixes/Improvements:** Fix the user pin and enable the maps recenter properties.

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #118 
- [] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached (if applicable).

##  Testing
- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [ ] Added units tests for this

If you haven't added tests, please create a new issue and assign it.

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
+            (180 / Math.PI) // Adjust for latitude
+
+    val southwest = LatLng(center.latitude - latDelta, center.longitude - lngDelta)
+    val northeast = LatLng(center.latitude + latDelta, center.longitude + lngDelta)
+
+    return RectangularBounds.newInstance(southwest, northeast)
+  }
+
   override fun searchPlaces(
       query: String,
```
Comment (apana03) [inline app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt:43]: Can you please add documentation for this class. We should make an effort on this matter as it came up in the individual feedback for M1

File: app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt [hunk 2]
```diff
@@ -107,7 +107,7 @@ class SearchScreenTest {
   @Test
   fun testNoResultsFoundPlaces() {
     val searchQuery = "test"
-    `when`(placesRepository.searchPlaces(eq(searchQuery), any(), any())).thenAnswer {
+    `when`(placesRepository.searchPlaces(eq(searchQuery), any(), any(), any())).thenAnswer {
       val onSuccess = it.arguments[1] as (List<Place>) -> Unit
       onSuccess(emptyList())
     }
```
Comment (apana03) [inline app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt:113]: Can we do more thorough testing? If possible I suggest testing the different possibilities concerning the location permissions (the popup, what happens when the location is not enabled, etc ...)

**Review by apana03 — CHANGES_REQUESTED** [review]
Nice work! This is a really useful feature for our app! However I fell like testing should be improved if possible.

File: app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt (hunk 1 above)
Comment (g0g05arui) [reply app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt:43]: This class was written by you though, but I will create the documentation for this.

Comment (apana03) [reply app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt:43]: I have another issue regarding this class #103. The comment just included the function you completed. The documentation for the rest of the class I will do when I raise the PR for the previous issue : )

File: app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt (hunk 2 above)
Comment (g0g05arui) [reply app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt:113]: I thought about this and for the permissions it is really trick to create tests as I am using a system activity that can't be mocked.

I think the current tests are sufficient for the search as the location doesn't matter in the tests as we are mocking that anyways.

File: app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt (hunk 1 above)
Comment (g0g05arui) [reply app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt:43]: Ok, I misunderstood your comment. Will add some context only for the new function

Comment (g0g05arui) [reply app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt:51]: Done

File: app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt (hunk 2 above)
Comment (g0g05arui) [reply app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt:113]: After we finish this feature completely (#103 is solved) I think we can add tests for the whole repository. Wdyt?

Comment (apana03) [reply app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt:113]: Oh ok. Then leave it as is, if our overall line coverage is above 80% then it is fine.

**Review by apana03 — APPROVED** [review]
Looks good, nice work Mihai!

**Pull request merged**